toolkit/stable_diffusion_model.py

```diff
@@ -1694,6 +1713,27 @@ def scale_model_input(model_input, timestep_tensor):
                             raise ValueError(
                                 f"Batch size of latents {latent_model_input.shape[0]} must be the same or half the batch size of timesteps {timestep.shape[0]}")
 
+            if self.controlnet and edge_maps is not None:
+                # Ensure edge_maps are on the correct device and dtype
+                edge_maps = edge_maps.to(self.device_torch, dtype=self.torch_dtype)
+                
+                # If we're doing classifier free guidance, we need to repeat the control image
+                if do_classifier_free_guidance:
+                    edge_maps = torch.cat([edge_maps] * 2, dim=0)
+                
+                # Get the additional residuals from the controlnet
+                down_block_res_samples, mid_block_res_sample = self.controlnet(
+                    latent_model_input,
+                    timestep,
+                    encoder_hidden_states=text_embeddings.text_embeds,
+                    controlnet_cond=edge_maps,
+                    return_dict=False,
+                )
+                
+                # Add the residuals to the kwargs
+                kwargs['down_block_additional_residuals'] = down_block_res_samples
```

Comment:
again, dont use kwargs for it. i gave you reference to the code of the pipeline and mentioned in which line you can look to see how it's added to the model inputs. and its not done using kwargs.